waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -179,19 +180,21 @@ proc addPeer*(pm: PeerManager, remotePeerInfo: RemotePeerInfo, proto: string) =
     # Do not attempt to manage our unmanageable self
     return
 
-  debug "Adding peer to manager", peerId = remotePeerInfo.peerId, addr = remotePeerInfo.addrs[0], proto = proto
-
-  # ...known addresses
-  for multiaddr in remotePeerInfo.addrs:
-    pm.peerStore[AddressBook][remotePeerInfo.peerId] = pm.peerStore[AddressBook][remotePeerInfo.peerId] & multiaddr
-
   # ...public key
   var publicKey: PublicKey
   discard remotePeerInfo.peerId.extractPublicKey(publicKey)
 
+  if pm.peerStore[AddressBook][remotePeerInfo.peerId] == remotePeerInfo.addrs and
+     pm.peerStore[KeyBook][remotePeerInfo.peerId] == publicKey:
+    # Peer already managed
+    return
```

Comment:
For readability reasons, this complex if statement condition should be replaced by a `pm.addressBook.isPeerManaged()` proc returning a boolean. If you do that, the comment on line 189 will be unnecessary.

Reply:
> Why thats not part of remotePeerInfo? well perhaps it should, but beyond this PR.

🤷🏼 

If I were working in Waku's peer management, I would modify the `RemotePeerInfo` type and add it. This is a fundamental issue that should be fixed if it helps improve code readability and reduces the cognitive load of the code.

As a temporary solution until the remote peer type includes the public key, I don't see any problem with extracting the key twice.